src/stream/src/executor/approx_percentile/global.rs

```diff
@@ -52,9 +52,10 @@ impl<S: StateStore> GlobalApproxPercentileExecutor<S> {
         // Initialize state
         let mut input_stream = self.input.execute();
         let first_barrier = expect_first_barrier(&mut input_stream).await?;
-        let mut state = self.state;
-        state.init(first_barrier.epoch).await?;
+        let first_epoch = first_barrier.epoch;
         yield Message::Barrier(first_barrier);
+        let mut state = self.state;
+        state.init(first_epoch).await?;
```

Comment:
Is it possible to have a declmacro to enforce the order if yielding and using the first barrier? Like

```rust
handle_first_barrier!(&mut input_stream, |barrier| {
    self.state.init(barrier.epoch).await?;
});
```

which internally calls `expect_first_barrier` + `yield Barrier` + the provided init block.

Reply:
It's difficult, because `yield barrier` in the custom macro will fail to compile in the `try_stream` proc macro. 

> Although in this PR we seem correctly changed all occurrences, it looks still dangerous to me that some other developer can easily break the order.

I think it's fine, because if the order is not followed, deadlock must happen. As long as the newly developed features are covered in CI test, which we are supposed to do, we can be confident that the order is correctly handled, and otherwise something must be wrong.

Reply:
OK, then maybe we can add some notes on this in the doc of `init_epoch`.